pallets/committee/src/lib.rs

```diff
@@ -380,6 +425,34 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        #[pallet::weight(10_000)] // TODO: Set weights
+        /// Propose a new constituent to the constituent committee
+        ///
+        /// * This call only can be called by `MemberType::Council`
+        /// * This call will propose a call which will rewrite the
+        ///   storage value of `Members`.
+        ///
+        /// ## NOTE
+        ///
+        /// This function contains a HACK, `frame_system::set_storage` directly.
+        pub fn propose_constituent(
+            origin: OriginFor<T>,
+            constituent: AccountIdFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_member(origin.clone(), true)?;
+            Self::propose(
+                origin,
+                Box::new(
+                    SystemCall::set_storage(Vec::from([Self::add_constituent_encoded(
+                        constituent,
+                    )?]))
+                    .into(),
+                ),
+            )?;
+
+            Ok(().into())
+        }
```

Comment:
After thinking about it again, I think we got the chain of events wrong. You're right we can't create an `Action` type on the fly because this type is configured in the runtime. So this will not be necessary anymore, instead we only require an `add_constituent` extrinsic.

Reply:
Considered both an `extrinsic` and a `call` before, this `propose_constituent` is for the UX, a sugar for simplifying this action.

But due to the origin issue, I agree with changing it to an extrinsic now.